src/components/tables/Table.tsx

```diff
@@ -0,0 +1,337 @@
+            {collapsibleData}
+          </TableCollapsibleRowConnected>
+        )
+        : null;
+
+      return (
+        <TableRowWrapper key={rowWrapperId}>
+          <TableHeadingRowConnected
+            id={headingAndCollapsibleId}
+            key={headingAndCollapsibleId}
+            tableId={this.props.id}
+            hide={this.props.tableState.isLoading || this.props.tableState.isInError}
+            isCollapsible={!!collapsibleData}
+            onClickCallback={(e: React.MouseEvent<any>) =>
+              this.props.onRowClick(this.props.getActions && this.props.getActions(rowData, this.props))
+            }>
+            {tableHeadingRowContent}
+          </TableHeadingRowConnected>
+          {collapsibleRow}
+        </TableRowWrapper>
+      );
+    });
+  }
+
+  buildBlankSlate(): JSX.Element {
+    const { tableState } = this.props;
+    const tableData = tableState && tableState.data || this.props.initialTableData;
+    const {
+      noResults,
+      noResultsOnFilterOrPredicates,
+      noResultsOnError,
+    } = this.props.blankSlates;
+
+    let blankSlatePropsToUse: IBlankSlateProps;
+
+    if (tableData.displayedIds.length
+      || _.isEmpty(this.props.blankSlates)
+      || this.props.tableState.isLoading) {
+      return null;
+    }
+
+    if (tableState
+      && (tableState.filter || _.some(tableState.predicates, (value: any) => !_.isUndefined(value)))) {
+      blankSlatePropsToUse = noResultsOnFilterOrPredicates || noResults;
+    } else if (tableState && tableState.isInError) {
+      blankSlatePropsToUse = noResultsOnError || noResults;
+    } else {
+      blankSlatePropsToUse = noResults;
+    }
+
+    return <BlankSlate {...blankSlatePropsToUse} />;
+  }
+
+  hideSectionOnLoading() {
+    return this.props.tableState.isLoading ? { display: 'none' } : undefined;
+  }
+
+  buildNavigation(): JSX.Element {
+    const tableData = this.props.tableState && this.props.tableState.data || this.props.initialTableData;
+
+    return !this.props.noNavigation ? (
+      <NavigationConnected
+        {...this.props.navigation}
+        totalEntries={tableData.totalEntries}
+        totalPages={tableData.totalPages}
+        id={getTableChildComponentId(this.props.id, TableChildComponent.NAVIGATION)}
+        loadingIds={[`loading-${getTableChildComponentId(this.props.id, TableChildComponent.NAVIGATION)}`]}
+        currentPage={this.props.tableState.page} />
+    ) : null;
+  }
+
+  buildLastUpdated(): JSX.Element {
+    return <LastUpdatedConnected
+      label={this.props.lastUpdatedLabel}
+      id={getTableChildComponentId(this.props.id, TableChildComponent.LAST_UPDATED)} />;
+  }
+
+  render() {
+    return (
+      <div>
+        {this.buildActionBar()}
+        <table className='mod-collapsible-rows'>
+          {this.buildTableHeader()}
+          {this.buildTableBody()}
+        </table>
+        {this.buildLoadingTable()}
+        {this.buildBlankSlate()}
```

Comment:
shouldn't the blankslate and loading be in the table?

Reply:
loadingTable won't be there anymore, blankSlate could go anywhere